src/parallelization/hpxsimulator.cpp

```diff
@@ -6,12 +6,16 @@
 namespace LibGeoDecomp {
 namespace HpxSimulatorHelpers {
 
+hpx::lcos::local::promise<void> initPromise;
```

Comment:
1. Who is responsible for resetting this promise if the user wants to create another HpxSimulator?
2. If the user is instantiating two HpxSimulators simultaneously this single static variable creates a race condition.